wiring/inc/spark_wiring_system.h

```diff
@@ -846,6 +875,37 @@ class SystemClass {
 
 #endif // HAL_PLATFORM_POWER_MANAGEMENT
 
+    int setControlRequestFilter(SystemControlRequestAclAction defaultAction, Vector<SystemControlRequestAclClass> filters = {}) {
+		// Add requests and their actions to the control filter
+        system_control_acl default_action = (system_control_acl)static_cast<typename std::underlying_type<SystemControlRequestAclAction>::type>(defaultAction);
+        system_control_filter* first = nullptr;
+        system_control_filter* current = nullptr;
+
+        for(int i=0; i<filters.size(); i++) {
+            if (first == nullptr) {
+                first = (system_control_filter*)malloc(sizeof(system_control_filter));
+                if (!first) {
+                    return SYSTEM_ERROR_NO_MEMORY;
+                }
+                first->req = filters[i].type();
+                first->action = (system_control_acl)static_cast<typename std::underlying_type<SystemControlRequestAclAction>::type>(filters[i].action());
+                LOG(TRACE, "TP1 - action: %d", first->action);
+                first->next = nullptr;
+                current = first;
+            } else {
+                system_control_filter* temp = (system_control_filter*)malloc(sizeof(system_control_filter));
+                current->next = temp;
+                temp->req = filters[i].type();
+                temp->action = (system_control_acl)static_cast<typename std::underlying_type<SystemControlRequestAclAction>::type>(filters[i].action());
+                LOG(TRACE, "TP2 - action: %d", temp->action);
+                temp->next = nullptr;
+                current = temp;
+            }
+        }
+
+        return system_ctrl_add_request_filter(default_action, first, nullptr);
```

Comment:
Memory allocated for list of filters is never freed. As proposed in `system_ctrl_add_request_filter`, take ownership of the linked list of filters in the SystemControl. Here, free memory only on errors.